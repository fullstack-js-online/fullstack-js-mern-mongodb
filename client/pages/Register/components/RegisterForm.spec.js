import React from 'react'
import { mount } from 'enzyme'

import RegisterForm from './RegisterForm'

describe('The <RegisterForm> component', () => {
  const props = {
    initialValues: {
      email: '',
      name: '',
      password: ''
    },
    onSubmit: jest.fn(),
    validationSchema: {}
  }

  it('should match snapshot', () => {
    expect(mount(<RegisterForm {...props} />)).toMatchSnapshot()
  })
})
