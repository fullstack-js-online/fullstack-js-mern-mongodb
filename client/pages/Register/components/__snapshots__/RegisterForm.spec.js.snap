// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <RegisterForm> component should match snapshot 1`] = `
<RegisterForm
  initialValues={
    Object {
      "email": "",
      "name": "",
      "password": "",
    }
  }
  onSubmit={[MockFunction]}
  validationSchema={Object {}}
>
  <div
    className="container mx-auto flex justify-center"
  >
    <div
      className="max-w-md-1/2 mx-3 my-12 w-full"
    >
      <h1
        className="text-center font-primary font-semibold text-gold"
      >
        Register
      </h1>
      <div
        className="bg-white shadow-md my-12 p-16 rounded w-full"
      >
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "email": "",
              "name": "",
              "password": "",
            }
          }
          isInitialValid={false}
          onSubmit={[MockFunction]}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={Object {}}
        >
          <form
            onSubmit={[Function]}
          >
            <Textbox
              data-testid="name-field"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your name"
              type="name"
              value=""
            >
              <div
                className="mb-5"
              >
                <input
                  className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter your name"
                  type="name"
                  value=""
                />
              </div>
            </Textbox>
            <Textbox
              data-testid="email-field"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your email"
              type="email"
              value=""
            >
              <div
                className="mb-5"
              >
                <input
                  className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter your email"
                  type="email"
                  value=""
                />
              </div>
            </Textbox>
            <Textbox
              data-testid="password-field"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your password"
              type="password"
              value=""
            >
              <div
                className="mb-5"
              >
                <input
                  className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter your password"
                  type="password"
                  value=""
                />
              </div>
            </Textbox>
            <Button
              data-testid="submit-button"
              loading={false}
              type="submit"
            >
              <button
                className="focus:outline-none rounded-sm hover:bg-emerald-light px-4 py-5 w-full text-white text-xl bg-emerald"
                disabled={false}
                type="submit"
              >
                Submit
              </button>
            </Button>
          </form>
        </Formik>
      </div>
    </div>
  </div>
</RegisterForm>
`;
